rr_common/src/planner/planner.h

```diff
@@ -6,6 +6,7 @@
 #include <cstring>
 #include <string>
 #include <random>
+#include <algorithm>
```

Comment:
Try to limit the scope of include statements. Only include headers in the files that directly use them.
In this case, you use the contents of `<algorithm>` in `panner.cpp` but include it in `planner.h`. The side effect here is that any files that include `planner.h` now have to include `<algorithm>` even though they won't be using it directly.